### PR TITLE
schema-bench package part2: add `wordOverlapMatch`  for name similarity

### PR DESCRIPTION
**Pull request by hoshinotsuyoshi**
## Why is this change needed?

This PR introduces and refines the `wordOverlapMatch` function, which performs lexical similarity matching between reference and predicted schema names using word overlap.
It is designed to improve the robustness and coverage of the schema evaluation framework by enabling:

* Token-based name comparison with stop word removal
* Case-insensitive matching
* Avoidance of duplicate candidate assignments

Additionally, I intentionally did **not** introduce the legacy Longest Common Substring (LCS) logic in order to **simplify** the implementation and clarify the matching criteria, as I found very few practical cases where it was actually useful.   https://github.com/liam-hq/liam/pull/2160/commits/23a6b1e8ccb3c942c51335364bc8fda98749b489

---

## What would you like reviewers to focus on?

* Whether the current word-based matching logic is sufficient and intuitive for lexical schema name similarity
* If the test coverage appropriately handles expected edge cases (e.g., case insensitivity, stop words, duplicate candidates)
* Confirmation that removing the LCS logic does not regress meaningful matching behavior in any known scenarios

### 🤖 Generated by PR Agent at 541deac2486eba0561361af14e6fae038479e42e

- Add `wordOverlapMatch` function for lexical schema name similarity
- Implement word tokenization with stop word removal
- Support case-insensitive matching and duplicate prevention
- Export new functionality from package entrypoint

## Detailed Changes

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Export word overlap matcher</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/internal-packages/schema-bench/src/index.ts

- Export `wordOverlapMatch` module from package entrypoint

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/2160/files#diff-39224dc4d7d8dacdf38dc7815ba6cbae9a14b4c5f6613eaa41773ff05a3c4bb0">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Add module index file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/internal-packages/schema-bench/src/wordOverlapMatch/index.ts

- Create module index file for `wordOverlapMatch`

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/2160/files#diff-4353fa65a1c7d0d9d7c9b011c0a5c3c1917b37428c5635a51f361c87ccf60295">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>wordOverlapMatch.ts</strong><dd><code>Implement word overlap matching logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.ts

<li>Implement <code>wordOverlapMatch</code> function for lexical similarity<br> <li> Add word tokenization with stop word filtering<br> <li> Use Set for O(1) candidate lookup optimization<br> <li> Support case-insensitive matching and prevent duplicate assignments

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/2160/files#diff-7a3b4d7951654a4ad39490b039216ab4aa95c62bf7bdc7d7f340eec827f6b476">+71/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>wordOverlapMatch.test.ts</strong><dd><code>Add comprehensive test coverage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts

<li>Test word overlap matching with various scenarios<br> <li> Cover case insensitivity, stop words, and duplicate prevention<br> <li> Test edge cases like no shared words and existing mappings

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/2160/files#diff-9c85a7f2584947fb83e31edf9e437336d33e95dd4142e9d7b3998f95ab4aecbf">+64/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Update documentation for word overlap</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/internal-packages/schema-bench/README.md

<li>Update documentation to reflect implemented <code>wordOverlapMatch</code><br> <li> Remove reference to unimplemented LCS logic

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/2160/files#diff-7ed1b3f08e940e63c28a97bc23bfdba20f244b2033d1445c9047954b756e47f3">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

## Additional Notes

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ */
+export function wordOverlapMatch(
+  references: string[],
+  candidates: string[],
+  mapping: Mapping,
+): void {
+  for (const referenceName of references) {
+    if (!(referenceName in mapping)) {
+      for (const predictName of candidates) {
+        if (!Object.values(mapping).includes(predictName)) {
```
[the remark this answers is not in the document]
Comment (hoshinotsuyoshi) [reply frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.ts]: fixed https://github.com/liam-hq/liam/pull/2160/commits/be61b6de9aa77555405351fe61db5e52ecd459c7

https://github.com/liam-hq/liam/pull/2160/commits/541deac2486eba0561361af14e6fae038479e42e

File: frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * Simple stop words list
+ */
+const STOP_WORDS = new Set(['the', 'a', 'an', 'of', 'record'])
+
+/**
+ * Convert string to word set: tokenize, lowercase, and remove stop words
+ */
+function toWordSet(str: string): Set<string> {
+  return new Set(
+    str
```
[the remark this answers is not in the document]
Comment (hoshinotsuyoshi) [reply frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.ts:27]: I think a bit complex implementation needed .
Added note on toWordSet limitations and possible improvements. thx!  https://github.com/liam-hq/liam/pull/2160/commits/6fc82ad34cd53cd3b211fa80af52aea6c85d76c8

File: frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts
Comment (MH4GF) [inline frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts:1]: That's easy to understand!

**Review by MH4GF — APPROVED** [review]
Nice work!

File: frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+describe('wordOverlapMatch', () => {
+  it('maps similar names by word overlap', () => {
+    const reference = ['Payment Record', 'Medical Claim', 'User Info']
+    const predict = ['Payment', 'MedicalRecord', 'User Information']
+    const mapping: Record<string, string> = {}
+    wordOverlapMatch(reference, predict, mapping)
+    expect(mapping).toEqual({
+      'Payment Record': 'Payment',
+      // 'Medical Claim': 'MedicalRecord',
```
Comment (FunamaYukina) [inline frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts]: [nits]
I noticed there is a commented-out code.🙏

Comment (hoshinotsuyoshi) [reply frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts]: That's right. I was trying to indicate that it's not included, but that seems vague. I'll re-examine my comment a bit! 🙏

Comment (FunamaYukina) [reply frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts]: Oh, I understand now. Thank you!

Comment (hoshinotsuyoshi) [reply frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts]: pushed a commit 🙏  https://github.com/liam-hq/liam/pull/2160/commits/c59bb730a051eba78ee47a60c5f32f58080c6794

Comment (FunamaYukina) [reply frontend/internal-packages/schema-bench/src/wordOverlapMatch/wordOverlapMatch.test.ts]: Thank you so much!✨
